GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -9,17 +9,194 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    
+    var seasons: [[GOTEpisode]] = GOTEpisode.allSeasons
+    
+    var filteredSeasons: [[GOTEpisode]] {
+        get {
+            guard let searchString = searchString else { return seasons }
+            
+            guard searchString != "" else { return seasons}
+            
+            if let scopeArrTitles = mySearchBar.scopeButtonTitles {
+                let currentIndex = mySearchBar.selectedScopeButtonIndex
+                let selectedStr = scopeArrTitles[currentIndex]
+                
+                return GOTEpisode.getFilteredResults(selectedStr: selectedStr, searchText: searchString)
+            }
+            
+            return seasons
+        }
+    }
+    
+    var searchString: String? = nil {
+        didSet {
+            self.myTableView.reloadData()
+        }
+    }
+    
+    @IBOutlet weak var myTableView: UITableView!
+    
+    @IBOutlet weak var mySearchBar: UISearchBar!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        configureTableView()
+        configureSearchBar()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
+    
+    func configureTableView() {
+        myTableView.dataSource = self
+        myTableView.delegate = self
+        myTableView.rowHeight = 80
+        myTableView.sectionHeaderHeight = 40
+        myTableView.tableFooterView = UIView()
+    }
+    
+    func configureSearchBar() {
+        mySearchBar.delegate = self
+    }
+    
+    
+
+
+}
+
+extension ViewController: UITableViewDataSource, UITableViewDelegate {
```

Comment:
Wow, awesome job making the extensions for the tableview. It makes everything easier to find and navigate to